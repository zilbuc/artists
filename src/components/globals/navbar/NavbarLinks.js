import React, { Component } from 'react';
import { Link } from 'gatsby';
import styled from 'styled-components';
import { styles } from '../../../utils';

class NavbarLinks extends Component {

  state = {
    links: [
      {
        id: 0,
        path: '/',
        name: 'home'
      },
      {
        id: 1,
        path: '/about/',
        name: 'about'
      },
      {
        id: 2,
        path: '/contact/',
        name: 'contact'
      }
    ]
  };

  render() {
    return (
      <LinksWrapper data-testid='links-wrapper' open={this.props.navbarOpen}>
        {
          this.state.links.map(item => {
            return (
              <li data-testid={`testid-link${item.id}`} key={item.id}><Link to={item.path} className='nav-link'>{item.name}</Link></li>
            )
          })
        }
      </LinksWrapper>
    );
  }

}

const LinksWrapper = styled.ul`
  height: ${props => (props.open ? '120px' : '0px')};
  overflow: hidden;
  padding-left: 0;
  margin: 0;
  ${styles.transObject({})};
  li {
    list-style-type: none;
  }
  .nav-link {
    display: block;
    text-decoration: none;
    padding: 0.5rem 1rem;
    color: ${styles.colors.mainGrey};
    font-size: 1.2rem;
    font-weight: 700;
    text-transform: capitalize;
    cursor: pointer;
    ${styles.transDefault};
    &:hover {
      background: ${styles.colors.mainGrey};
      color: ${styles.colors.mainYellow};
      padding-left: 1.3rem;
    }
  }
  @media (min-width: 992px) {
    height: auto;
    display: flex;
    margin: 0 auto;
    .nav-link:hover {
      background: ${styles.colors.mainWhite};
      padding-left: 1rem;
    }
  }
`;

export default NavbarLinks;
