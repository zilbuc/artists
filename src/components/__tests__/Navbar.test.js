import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import 'jest-dom/extend-expect';
import Navbar from '../globals/navbar/Navbar';
import NavbarHeader from '../globals/navbar/NavbarHeader';
import NavbarIcons from '../globals/navbar/NavbarIcons';
import NavbarLinks from '../globals/navbar/NavbarLinks';

afterEach(cleanup);

describe('Navbar', () => {
  it('renders NavbarHeader correctly', () => {
    const { asFragment } = render(<NavbarHeader />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays NavbarLinks links', () => {
    const { getByTestId, getByText } = render(<NavbarLinks />);

    expect(getByTestId('testid-link0')).toHaveTextContent('home');
    expect(getByText('about')).toBeVisible();
    expect(getByText('contact')).toBeInTheDocument();
  });

  it('displays NavbarIcons icons', () => {
    const { getByTestId, queryByTestId } = render(<NavbarIcons />);

    const ancestor = queryByTestId('icon-wrapper');
    const descendant = queryByTestId('testid-icon2');
    const nonExistantElement = queryByTestId('i-do-not-exist');

    expect(nonExistantElement).not.toBeInTheDocument();
    expect(getByTestId('testid-icon1')).toBeInTheDocument();
    expect(ancestor).toContainElement(descendant);
    expect(getByTestId('testid-icon3')).toBeInTheDocument();
  });

  it('has click icon that shows/hides NavbarLinks', () => {
    const { getByText, getByTestId } = render(<Navbar />);

    expect(getByTestId('navbar-icon')).toBeVisible();
    expect(getByTestId('links-wrapper')).toHaveStyle('height: 0px');

    fireEvent.click(getByTestId('navbar-icon'));

    expect(getByTestId('links-wrapper')).toHaveStyle('height: 120px');
  });
});
