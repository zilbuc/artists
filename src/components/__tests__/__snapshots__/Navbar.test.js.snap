// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders NavbarHeader correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-gZMcBi iZoYzb"
  >
    <a
      href="/"
    >
      <img
        alt="company name"
        src="test-file-stub"
      />
    </a>
    <svg
      class="toggle-icon"
      data-testid="navbar-icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M160 84V44c0-8.837 7.163-16 16-16h256c8.837 0 16 7.163 16 16v40c0 8.837-7.163 16-16 16H176c-8.837 0-16-7.163-16-16zM16 228h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 256h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm160-128h256c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H176c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      />
    </svg>
  </div>
</DocumentFragment>
`;
